test/functional/Completion.hs

```diff
@@ -116,7 +118,7 @@ tests = testGroup "completions" [
          let te = TextEdit (Range (Position 0 13) (Position 0 31)) "NoOverload"
          _ <- applyEdit doc te
 
-         compls <- getCompletions doc (Position 0 24)
+         compls <- getCompletions doc (Position 0 23)
          let item = head $ filter ((== "NoOverloadedStrings") . (^. label)) compls
```

Comment:
So before did we just generate basically _all_ the completions, because we weren't applying the prefix? If so, do we need a test that hte number of compls is < 5 or something reasonable?

Reply:
Previously, the test was generating completions for the empty string, which means all the possible pragmas. Since there is more pragmas than *maxCompletions*, the test could no longer pass.

Reply:
From my perspective this test is meant to verify that we generate the `NoOverload` completion, and other checks belong elsewhere

Reply:
If you generate completions for everything, they aren't that useful, so I consider it important to have some kind of check enforcing some kind of restricted set.